src/Compilers/VisualBasic/Portable/Syntax/InternalSyntax/SyntaxNodeExtensions.vb

```diff
@@ -592,16 +592,51 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return skippedTriviaBuilder.GetTriviaList()
         End Function
 
+        ''' <summary>
+        ''' Add all the tokens in this node and children to the build token list builder. While doing this, add any
+        ''' diagnostics not on tokens to the given diagnostic info list.
+        ''' </summary>
+        Friend Sub CollectConstituentTokensAndDiagnostics(
+                this As GreenNode,
+                tokenListBuilder As SyntaxListBuilder(Of SyntaxToken),
+                nonTokenDiagnostics As IList(Of DiagnosticInfo))
+            If this Is Nothing Then
+                Return
+            End If
+
+            If this.IsToken Then
+                tokenListBuilder.Add(DirectCast(this, SyntaxToken))
+                Return
+            End If
+            ' This implementation is overridden for tokens; this is the implementation for non-token nodes.
```

Comment:
Is this comment still correct? It looks like the token case is immediately above.